django_opensearch_dsl/documents.py

```diff
@@ -115,14 +105,41 @@ def get_indexing_queryset(
             if verbose:
                 stdout.write(f"{action} {model}: {round(i / count * 100)}% ({self._eta(start, done, count)})\r")
 
-            for obj in qs[i : i + chunk_size]:
+            for obj in qs.iterator(chunk_size=chunk_size):
                 done += 1
+                if done % chunk_size == 0:
+                    stdout.write(f"{action} {model}: {round(done / count * 100)}% ({self._eta(start, done, count)})\r")
                 yield obj
 
-            i = min(i + chunk_size, count)
-
-        if verbose:
-            stdout.write(f"{action} {count} {model}: OK          \n")
+    def get_indexing_queryset(
+        self,
+        db_alias: str = None,
+        verbose: bool = False,
+        filter_: Optional[Q] = None,
+        exclude: Optional[Q] = None,
+        count: int = None,
+        action: OpensearchAction = OpensearchAction.INDEX,
+        stdout: io.FileIO = sys.stdout,
+        batch_size: int = None,
+    ) -> Iterable:
+        """Divide the queryset into chunks."""
+        qs = self.get_queryset(db_alias=db_alias, filter_=filter_, exclude=exclude, count=count)
+        if self._order_indexing_queryset and not qs.query.is_ordered:
+            qs = qs.order_by("pk")
+        if batch_size:
+            # In order to avoid loading big querysets into memory or
+            # loading them in temporary tables in the database,
+            # we have the possibility to divide the queryset using batch_size.
```

Comment:
I see no point in not batching every time since not batching at all will probably lead to an error on a large dataset.

A reasonable default for `batch_size` would be good